SDWebImage/SDWebImageManager.m

```diff
@@ -270,11 +256,54 @@ - (void)diskImageExistsForURL:(nullable NSURL *)url
                         [self callCompletionBlockForOperation:strongSubOperation completion:completedBlock image:downloadedImage data:downloadedData error:nil cacheType:SDImageCacheTypeNone finished:finished url:url];
                     }
                 }
-
+                
                 if (finished) {
                     [self safelyRemoveOperationFromRunning:strongSubOperation];
                 }
-            } context:context];
+            };
+            
+            id<SDWebImageOperation> loadOperation;
+            // Firstlly check loading image instance
+            if ([self.imageLoader respondsToSelector:@selector(loadImageWithURL:options:progress:completed:context:)]) {
+                loadOperation = [self.imageLoader loadImageWithURL:url options:options progress:progressBlock completed:imageLoadCompletion context:context];
+            }
+            // Secondlly check loading image data
+            if (!loadOperation) {
+                if ([self.imageLoader respondsToSelector:@selector(loadImageDataWithURL:options:progress:completed:context:)]) {
+                    // Dispatch the decoding progress into the coder queue
+                    strongOperation.coderQueue = dispatch_queue_create("com.hackemist.SDWebImageManagerCoderQueue", DISPATCH_QUEUE_SERIAL);
+                    loadOperation = [self.imageLoader loadImageDataWithURL:url options:options progress:progressBlock completed:^(NSData * _Nullable data, NSError * _Nullable error, BOOL finished) {
+                        __strong typeof(weakSubOperation) strongSubOperation = weakSubOperation;
+                        if (!strongSubOperation) {
+                            return;
+                        }
+                        if (!error) {
+                            dispatch_async(strongSubOperation.coderQueue, ^{
+                                UIImage *loadedImage = [[SDWebImageCodersManager sharedManager] decodedImageWithData:data];
```

Comment:
Acutally, here is not finished because we supports progressive image loading.

One solution, it's to leave this for developer to use the `loadImageWithURL:` one. Or I can continue to implement it to allow `loadImageDataWithURL:` progressive loading. (Actually, `copy` the code in `SDWebImageDownloadOperation`. Or maybe a better way ?).